libs/design-system/src/pagination/usePaginationState.ts

```diff
@@ -35,13 +34,11 @@ export const usePaginationState = ({
       return;
     }
 
-    setSearchParams(
-      {
-        [pageNumberParamName]: `${currentPageNumber}`,
-        [pageSizeParamName]: `${pageSize}`,
-      },
-      { replace: true }
-    );
+    const newSearchParams = new URLSearchParams(document.location.search);
+    newSearchParams.set(pageNumberParamName, `${currentPageNumber}`);
+    newSearchParams.set(pageSizeParamName, `${pageSize}`);
+
+    setSearchParams(newSearchParams, { replace: true });
```

Comment:
the `usePaginationState` and `useSearchState` are both separate hooks that both update the query parameters. This code ensures that they are not overridden but all combined together. Otherwise, if the `useEffect` in the `useSearchState` hook is called after, it will override all the pagination query params.